include/render/gles2.h

```diff
@@ -38,21 +38,49 @@ struct wlr_gles2_renderer {
 	} shaders;
 };
 
+enum wlr_gles2_texture_type {
+	WLR_GLES2_TEXTURE_GLTEX,
+	WLR_GLES2_TEXTURE_WL_DRM_GL,
+	WLR_GLES2_TEXTURE_WL_DRM_EXT,
+	WLR_GLES2_TEXTURE_DMABUF,
+};
+
 struct wlr_gles2_texture {
 	struct wlr_texture wlr_texture;
 
-	struct wlr_egl *egl;
-	GLuint tex_id;
-	const struct gles2_pixel_format *pixel_format;
+	struct wlr_gles2_renderer *renderer;
+	enum wlr_gles2_texture_type type;
+	int width, height;
+	bool has_alpha;
```

Comment:
Is this (and others here) a propery of GLES2 textures and not rather textures in general (wlr_texture)?